packages/flutter_web_plugins/lib/src/navigation/utils.dart

```diff
@@ -15,30 +13,30 @@ AnchorElement _urlParsingNode;
 String extractPathname(String url) {
   // TODO(mdebbar): Use the `URI` class instead?
   _urlParsingNode ??= AnchorElement();
-  _urlParsingNode.href = url;
-  final String pathname = _urlParsingNode.pathname;
+  _urlParsingNode!.href = url;
+  final String pathname = _urlParsingNode!.pathname ?? '';
```

Comment:
Is `?? ''` the right fallback? In the previous version `pathname` continued to be null. This also ties into the previous question about whether `pathname` should ever be null.

Reply:
In this case, pathname comes from dart:html `class AnchorElement`, and it's [defined as `String?`][pathname]. In my tests it's never null, in fact, if no pathname is supplied, in chrome it returns `'/'`. According to the spec, it should return `''`.

Looking at the next line of code (which I haven't touched), it seems that `pathname` will *never* be null, regardless of what the definition says. I'd say an empty string is a nice fallback.

_(Just to double check, you can't call isEmpty on null)_
```
Uncaught TypeError: Cannot read property 'get$isEmpty' of nullError: TypeError: Cannot read property 'get$isEmpty' of null
```

[pathname]: https://api.dart.dev/stable/2.10.3/dart-html/AnchorElement/pathname.html